cpp/src/arrow/vendored/musl/strptime.c

```diff
@@ -18,8 +18,10 @@
 #undef HAVE_LANGINFO
 
 #ifndef _WIN32
+# if !(defined(__ANDROID__) && __ANDROID_API__ < 26)
```

Comment:
Can we unify this condition?

```c
#if !defined(_WIN32) && !(defined(__ANDROID__) && __ANDROID_API__ < 26)
```

BTW, is Android API < 26 still supported?
It seems that Android API 25 is Android 7.1 based on https://developer.android.com/guide/topics/manifest/uses-sdk-element#api-level-table .
It seems that Android 11 or earlier reached EOL based on https://endoflife.date/android .

Reply:
that's a good point, we can remove this patch and probably remove it also upstream on the next release. It shouldn't be necessary as unsupported version, I suppose if we add an Android CI job might be worth validating